### PR TITLE
added useful changes and comments in countingSort.c

### PR DESCRIPTION
**Pull request by harshraj22**
This PR includes work done : 
- [x] Indented Code
- [x] Updated Code to be able to handle unexpected inputs
- [x] Added comments for better understanding of the code.

Issue #259

### CODE REVIEW

File: sorting/countingSort.c [hunk 1]
```diff
@@ -1,28 +1,45 @@
...
\ No newline at end of file
+#include <string.h>	// for memset
+#include <assert.h>
+
+int main()
+{
+	int i, n, l = 0;
+	scanf("%d", &n);
+
+	assert(n < 1e7);
```
Comment (danghai) [inline sorting/countingSort.c]: Why do you use a magic number? Simply use constant number for initial array

Comment (harshraj22) [reply sorting/countingSort.c]: The maximum length of array is fixed due to memory constraint of mordern day computer ( array occupies contigous memory block). So it is advisable not to try to create a huge sized array. The max size of locally defined array is roughly 1e7 ie 10^7 . [reference](https://discuss.codechef.com/t/maximum-size-of-an-array/3633)

Comment (danghai) [reply sorting/countingSort.c]: It just illustrates how counting sort works, so using constant number  (ex: 256) is good enough.

Comment (harshraj22) [reply sorting/countingSort.c]: @danghai Fine. Updated accordingly.

**Comment by ashwek** [comment]
@harshraj22 please resolve merge conflict

**Comment by ashwek** [comment]
@harshraj22 please resolve merge conflict

**Comment by harshraj22** [comment]
@ashwek Done.

**Pull request merged**